capi/geos_ts_c.cpp

```diff
@@ -1242,6 +1243,31 @@ extern "C" {
         });
     }
 
+    char
+    GEOSisSimpleDetail_r(GEOSContextHandle_t extHandle, const Geometry* g1, int returnAllPoints, Geometry** result)
+    {
+        return execute(extHandle, 2, [&]() {
+            geos::operation::valid::IsSimpleOp iso(g1);
+            iso.setFindAllLocations(returnAllPoints);
+
+            if (result != nullptr) {
+                *result = nullptr;
+            }
```

Comment:
```suggestion
            *result = nullptr;
```
No reason to call this with `result = nullptr`